src/datumaro/components/importer.py

```diff
@@ -197,7 +196,7 @@ def _change_context_root_path(context: FormatDetectionContext, path: str):
                 )
 
             for sub_dir in os.listdir(path):
-                if sub_dir.lower() in SUBSET_NAME_BLACKLIST:
+                if sub_dir.lower() not in ("train", "test", "val"):
```

Comment:
Why this change is needed?

Reply:
This is necessary to distinguish between two importers with unlimited hierarchy levels: `nested_imagenet` and `nested_imagenet_with_subset_dirs`. Without this distinction, we cannot determine whether a folder is a subset directory or a top-level class.

Reply:
I meant what is the reason that we should remove `SUBSET_NAME_BLACKLIST`. Is it impossible such as

```suggestion
from datumaro.util.definitions import SUBSET_NAME_BLACKLIST, SUBSET_NAME_WHITELIST
# SUBSET_NAME_WHITELIST = ("train", "test", "val")

                if sub_dir.lower() in SUBSET_NAME_BLACKLIST or sub_dir.lower() not in SUBSET_NAME_WHITELIST:
```

p.s. I saw `("train", "test", "val")` is used other place so it seems possible to share it by declaring it at `datumaro.util.definitions`

Reply:
If the whitelist condition is not fulfilled, the `sub_dir` is considered to belong to the blacklist, and vice versa. If the directory name is not on either list, it is treated as if it belongs to the blacklist and is not processed. This logically equals to process only whitelist condition.

Reply:
Ah ok then can you move it to `datumaro.util.definitions` to be shared in other source files such as `src/datumaro/plugins/data_formats/imagenet.py`